src/hooks/useRedirectLink.js

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react";
+import redirect from "../utils/routing/routing";
+
+/* This custom hook is meant to be used for redirecting
+on pages that should support redirect links */
+export default function useRedirectLink() {
+  const [existingRedirectLink, setExistingRedirectLink] = useState(null);
+
+  //check for existing redirect links
+  useEffect(() => {
+    const queryParameters = new URLSearchParams(window.location.search);
+    setExistingRedirectLink(queryParameters.get("redirectLink"));
```

Comment:
i think `existingRedirectLink` can also be a const. there is no need to have it state. or am I missing something?

Reply:
`existingRedirectLink` needs to have a state **because this state controls** how the `handleRedirectLinkOrGoTo` function behaves.
- If the `existingRedirectLink`'s state is `null`, the function redirects the user to the location provided in its argument. For example, `handleRedirectLinkOrGoTo("/articles")` will redirect the user to the articles page. 
- If the `existingRedirectLink`'s state is **not** `null`, the function redirects the user to the **location** from the `existingRedirectLink` instead and ignores the one provided in the argument.

If we defined `existingRedirectLink` as a const in the `useEffect`, it would be only accessible within the scope of this `useEffect` but not outside.